File: agent/nodes/fetchTickets.js
const { fetchTodoTickets } = require('../tools/jira');

/**
 * FETCH TICKETS node — pull TODO tickets from Jira.
 * @param {import('../state').AgentStateAnnotation} state
 */
async function fetchTicketsNode(state) {
  console.log('\n📋 NODE: fetchTickets — fetching TODO tickets from Jira...');

  try {
    const tickets = await fetchTodoTickets();
    console.log(`  ✅ Found ${tickets.length} ticket(s)`);
    tickets.forEach((t) => console.log(`    - ${t.key}: ${t.fields?.summary}`));

    return { ...state, tickets };
  } catch (err) {
    console.error('  ❌ fetchTickets failed:', err.message);
    return { ...state, tickets: [], error: `fetchTickets: ${err.message}` };
  }
}

module.exports = { fetchTicketsNode };

File: python-backend/nodes/fetch_tickets.py
from tools.jira import fetch_todo_tickets
from state import AgentState


async def fetch_tickets_node(state: AgentState) -> AgentState:
    print("\n📋 NODE: fetchTickets — fetching TODO tickets from Jira...")
    try:
        tickets = await fetch_todo_tickets()
        print(f"  ✅ Found {len(tickets)} ticket(s)")
        for t in tickets:
            print(f"    - {t['key']}: {t['fields']['summary']}")
        return {**state, "tickets": tickets}
    except Exception as e:
        print(f"  ❌ Failed to fetch tickets: {e}")
        return {**state, "tickets": [], "error": str(e)}
